osu.Game/Skinning/Skin.cs

```diff
@@ -293,20 +304,71 @@ private void applyMigration(SkinLayoutInfo layout, GlobalSkinnableContainers tar
                         c.Type.Name == nameof(DefaultComboCounter) ||
                         c.Type.Name == nameof(ArgonComboCounter)).ToArray();
 
-                    layout.Update(null, globalHUDComponents.Except(comboCounters).ToArray());
+                    hudLayout.Update(null, globalHUDComponents.Except(comboCounters).ToArray());
 
                     resources.RealmAccess.Run(r =>
                     {
                         foreach (var ruleset in r.All<RulesetInfo>())
                         {
-                            layout.Update(ruleset, layout.TryGetDrawableInfo(ruleset, out var rulesetHUDComponents)
+                            hudLayout.Update(ruleset, hudLayout.TryGetDrawableInfo(ruleset, out var rulesetHUDComponents)
                                 ? rulesetHUDComponents.Concat(comboCounters).ToArray()
                                 : comboCounters);
                         }
                     });
 
                     break;
                 }
+
+                case 2:
+                {
+                    // Legacy health bar is moved from the global HUD components target into per-ruleset.
+                    // On osu!mania, the health bar is moved from the HUD to Playfield target for stage-based positioning.
+                    if (resources == null)
+                        break;
```

Comment:
What is this check... checking for? It's completely inscrutable to me.

Reply:
Mostly ignore doing anything if we don't have the thing to do it, it only matters in tests where skins are instantiated without resources. This check existed before so I'm not sure why it's being asked in here.